src/test/java/BaseTest.java

```diff
@@ -1,18 +1,169 @@
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
 import org.openqa.selenium.firefox.FirefoxDriver;
-import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.UUID;
 
 
 public class BaseTest {
+    public WebDriver driver = null;
+    public String url = null;
+    public WebDriverWait wait = null;
+
+    public FluentWait fluentWait = null;
+    public Actions action = null;
+    public ThreadLocal<WebDriver> threadLocal = null;
+
+
+
+//      @BeforeSuite
+//    static void setupClass() {
+//        WebDriverManager.chromedriver().setup();
+//    }
+
+  //  @BeforeMethod
+    //@Parameters({"BaseUrl"})
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.